brush-core/src/builtins/mapfile.rs

```diff
@@ -93,46 +80,71 @@ impl builtins::Command for MapFileCommand {
 }
 
 impl MapFileCommand {
-    fn read_entries(
+    async fn read_entries(
         &self,
-        mut input_file: openfiles::OpenFile,
+        input_file: openfiles::OpenFile,
+        context: &mut commands::ExecutionContext<'_>,
     ) -> Result<variables::ArrayLiteral, error::Error> {
-        let mut entries = vec![];
-
         let orig_term_attr = setup_terminal_settings(&input_file)?;
```

Comment:
In trying this out a bit, I see a (preexisting) oddity in the terminal settings we go with here. It looks like backspace still does the normal backspace thing in `bash`'s version of `mapfile`, but I'm seeing it emit `^?` in `brush`. That's not specific to this PR or any of your changes, so we can just file a separate issue on it. I'm noting it here mostly so I don't forget to mention it.